mne/decoding/time_gen.py

```diff
@@ -607,84 +694,82 @@ def _fit_slices(clf, x_chunk, y, slices, cv):
     return estimators
 
 
-def _sliding_window(times, window_params, sfreq):
+def _sliding_window(times, window, sfreq):
     """Aux function of GeneralizationAcrossTime
 
-    Define the slices on which to train each classifier.
+    Define the slices on which to train each classifier. The user either define
+    the time slices manually in window['slices'] or s/he passes optional params
```

Comment:
This PR fixes a also bug with irregular time windows. I think it'd be cleaner to first fix the bug, add a deprecation decorator, and deprecate in the following release.